docs/configuration/features/image_features.md

```diff
@@ -1,76 +1,340 @@
+  - `interpolate`: Uses interpolation to resize images to the specified `width` and `height`
+
+### `infer_image_dimensions`
+
+If set, then the height and width of images in the dataset will be inferred from a sample of the first `n` images in the dataset. The size of `n` is set by the parameter `infer_image_sample_size`. Each image that doesn't conform to these dimensions will be resized according to `resize_method`. If set to `false`, then the height and width of images in the dataset will be specified by the user.
+
+This parameter will have no effect if `width` and `height` are specified.
+
+- Default: `true`
+
+### `infer_image_max_height`
+
+If `infer_image_dimensions` is set, this is used as the maximum height of the images in the dataset.
+
+- Default: 256
+
+### `infer_image_max_width`
+
+If `infer_image_dimensions` is set, this is used as the maximum width of the images in the dataset.
+
+- Default: 256
+
+### `infer_image_num_channels`
+
+If set, then the number of channels in the dataset is inferred from a sample of the first `n` images in the dataset. The size of `n` is set by the parameter `infer_image_sample_size`.
+
+- Default: `true`
+
+### `infer_image_sample_size`
+
+The sample size used for inferring dimensions of images in `infer_image_dimensions`.
+
+- Default: 100
+
+### `scaling`
+
+The scaling strategy for pixel values in the image.
+
+- Default: `pixel_normalization`
+- Options:
+  - `pixel_normalization`: Normalizes pixel values to be between 0 and 1 by dividing each pixel value by 255.
+  - `pixel_standardization`: Normalizes pixel values based on the mean and standard deviation of images in ImageNet.
+
+### `in_memory`
+
+Whether image dataset will reside in memory during the training process or will be dynamically fetched from disk (useful for large datasets). In the latter case a training batch of input images will be fetched from disk each training iteration.
+
+- Default: `true`
+
+### `num_processes`
+
+Specifies the number of processes to run for preprocessing images.
+
+- Default: 1
+
+!!! note
+    Depending on the application, it is preferrable not to exceed a size of `256 x 256`, as bigger sizes will, in most cases, not provide much advantage in terms of performance, while they will considerably slow down training and inference and also make both forward and backward passes consume considerably more memory, leading to memory overflows on machines with limited amounts of RAM or on GPUs with limited amounts of VRAM.
 
 Example of a preprocessing specification:
 
 ```yaml
 name: image_feature_name
 type: image
 preprocessing:
+  missing_value_strategy: fill_with_const
+  fill_value: 0.5
   height: 128
   width: 128
+  num_channels: 3
   resize_method: interpolate
   scaling: pixel_normalization
+  in_memory: true
+  num_processes: 4
 ```
 
-## Image Input Features and Encoders
+# Image Input Features and Encoders
```

Comment:
This is h2 in the other features docs, lets keep it that way for now and make a global style change later.

Reply:
Same as above, keeping the headers this way for now to avoid duplication of effort.